### PR TITLE
Config: Add forbiddenNoteIds to AppConfig

### LINKED ISSUE #519: Which forbidden note ids should we keep?
Which of these via default forbidden note ids should we keep? Which are unnecessary anymore?
I think we want to keep this config option around, since admins will possibly used this to accomplish things.

https://github.com/codimd/server/blob/3461993ee05d09106e276e606b143d472908b166/lib/config/default.js#L36

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
config, notesService

### Description
This PR adds the forbiddenNoteIds to the AppConfig.

Also uses this config in the notes service and the appropriate tests.

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### Related Issue(s)
fixes #519

### CODE REVIEW

**Review by InnayTool — APPROVED** [review]

**Review by davidmehren — COMMENTED** [review]
Looks good!

Currently, we only check for a forbidden id when creating a new note. We probably also want to do this when accessing already existing notes?

**Comment by DerMolly** [comment]
> Currently, we only check for a forbidden id when creating a new note. We probably also want to do this when accessing already existing notes?

Probably not wrong…

**Review by InnayTool — APPROVED** [review]

**Review by InnayTool — APPROVED** [review]

File: src/notes/notes.service.ts [hunk 1]
```diff
@@ -148,6 +161,15 @@ export class NotesService {
       `Trying to find note '${noteIdOrAlias}'`,
       'getNoteByIdOrAlias',
     );
+    if (this.appConfig.forbiddenNoteIds.includes(noteIdOrAlias)) {
+      this.logger.debug(
+        `A note with the alias '${noteIdOrAlias}' is forbidden by the administrator.`,
+        'createNote',
```
Comment (davidmehren) [inline src/notes/notes.service.ts:167]:
```suggestion
        'getNoteByIdOrAlias',
```

File: src/notes/notes.service.ts [hunk 2]
```diff
@@ -88,6 +92,15 @@ export class NotesService {
     ]);
     if (alias) {
       newNote.alias = alias;
+      if (this.appConfig.forbiddenNoteIds.includes(alias)) {
+        this.logger.debug(
+          `A note with the alias '${alias}' is forbidden by the administrator.`,
+          'createNote',
+        );
+        throw new ForbiddenIdError(
+          `A note with the alias '${alias}' is forbidden by the administrator.`,
```
Comment (davidmehren) [inline src/notes/notes.service.ts:101]:
```suggestion
          `Creating a note with the alias '${alias}' is forbidden by the administrator.`,
```

File: src/notes/notes.service.ts [hunk 3]
```diff
@@ -148,6 +161,15 @@ export class NotesService {
       `Trying to find note '${noteIdOrAlias}'`,
       'getNoteByIdOrAlias',
     );
+    if (this.appConfig.forbiddenNoteIds.includes(noteIdOrAlias)) {
+      this.logger.debug(
+        `A note with the alias '${noteIdOrAlias}' is forbidden by the administrator.`,
+        'createNote',
+      );
+      throw new ForbiddenIdError(
+        `A note with the alias '${noteIdOrAlias}' is forbidden by the administrator.`,
```
Comment (davidmehren) [inline src/notes/notes.service.ts:170]:
```suggestion
        `Accessing a note with the alias '${noteIdOrAlias}' is forbidden by the administrator.`,
```

**Review by davidmehren — CHANGES_REQUESTED** [review]

File: src/notes/notes.service.ts (hunk 1 above)
Comment (DerMolly) [reply src/notes/notes.service.ts]: fixed

File: src/notes/notes.service.ts (hunk 2 above)
Comment (DerMolly) [reply src/notes/notes.service.ts]: fixed

File: src/notes/notes.service.ts (hunk 3 above)
Comment (DerMolly) [reply src/notes/notes.service.ts]: fixed

**Review by davidmehren — APPROVED** [review]

**Pull request merged**